### ISSUE TITLE
Fails to install with pip. The source directory does not appear to contain CMakeLists.txt.

### ISSUE THREAD

**Issue by joeyli99** [pyscf/pyscf#1112]
I just simply run
`pip install pyscf`

`cmake -SC:\Users\gengp\AppData\Local\Temp\pip-install-7om9cog0\pyscf_5f5696130a59493d91d549229efb7b2f\pyscf\lib -Bbuild\temp.win-amd64-3.9\Release
CMake Error: The source directory "C:/Users/gengp/AppData/Local/Temp/pip-install-7om9cog0/pyscf_5f5696130a59493d91d549229efb7b2f/pyscf/lib" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.`

My cmake version is 3.21.4

**Comment by xubwa** [comment]
It seems that the path is from a windows machine, pyscf is not available on windows, but you can try to use windows subsystem of linux (wsl) instead, it mostly works fine there.

**Comment by ccoulombe** [comment]
The sdist on PyPI does not contains the [CMakeLists.txt](https://github.com/pyscf/pyscf/blob/master/pyscf/lib/CMakeLists.txt) file from in `pyscf/lib`. 

It then fails to build with the error: `pyscf/lib does not appear to contain CMakeLists.txt`


```
# sdist file downloaded from PyPI
$ tar -list -f pyscf-2.0.1.tar.gz | fgrep -i cmake
$ 
```

I had to download the archive from Github [`v2.0.1.tar.gz`](https://github.com/pyscf/pyscf/archive/refs/tags/v2.0.1.tar.gz):
```
$ tar -list -f v2.0.1.tar.gz | fgrep -i cmake
pyscf-2.0.1/pyscf/lib/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/agf2/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/ao2mo/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/cc/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-gnu
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-gnu-gpu
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-gnu-mkl
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-intelpython-gfortran
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-intelpython-ifort
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-nao-gnu
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-oberon.anaconda.gnu.mkl
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-oberon.anaconda.ifort.mkl
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-pgi-gpu
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-singularity.anaconda.gnu.mkl
pyscf-2.0.1/pyscf/lib/cmake_user_inc_examples/compile-atlas.sh
pyscf-2.0.1/pyscf/lib/dft/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/gto/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/mcscf/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/np_helper/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/pbc/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/ri/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/sparsetools/CMakeLists.txt
pyscf-2.0.1/pyscf/lib/vhf/CMakeLists.txt
```
With which I can successfully build the v2.0.1.

**Comment by pinskip** [comment]
I can confirm this problem with miniforge on Linux (conda version : 4.10.3).

The environment (conda list):

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2021.10.8            ha878542_0    conda-forge
h5py                      3.6.0                    pypi_0    pypi
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
libblas                   3.9.0           12_linux64_openblas    conda-forge
libcblas                  3.9.0           12_linux64_openblas    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 11.2.0              h1d223b6_11    conda-forge
libgfortran-ng            11.2.0              h69a702a_11    conda-forge
libgfortran5              11.2.0              h5c6108e_11    conda-forge
libgomp                   11.2.0              h1d223b6_11    conda-forge
liblapack                 3.9.0           12_linux64_openblas    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libopenblas               0.3.18          pthreads_h8fe5266_0    conda-forge
libstdcxx-ng              11.2.0              he4da1e4_11    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libzlib                   1.2.11            h36c2ea0_1013    conda-forge
ncurses                   6.2                  h58526e2_4    conda-forge
numpy                     1.21.4                   pypi_0    pypi
openssl                   3.0.0                h7f98852_2    conda-forge
pip                       21.3.1             pyhd8ed1ab_0    conda-forge
python                    3.10.0          h543edf9_3_cpython    conda-forge
python_abi                3.10                    2_cp310    conda-forge
readline                  8.1                  h46c0cb4_0    conda-forge
scipy                     1.7.3                    pypi_0    pypi
setuptools                59.2.0                   pypi_0    pypi
sqlite                    3.36.0               h9cd32fc_2    conda-forge
tk                        8.6.11               h27826a3_1    conda-forge
tzdata                    2021e                he74cb21_0    conda-forge
wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
zlib                      1.2.11            h36c2ea0_1013    conda-forge
```

**Comment by jamesETsmith** [comment]
@AlexandreF-1qbit, you likely aren't seeing this problem for those other Python versions because you aren't building them from source and are just using the prebuilt wheels. For Python 3.10, there is no prebuilt wheel and when you pip install it builds from source. In your python 3.10 environment you could pip install using github with `pip install git+https://github.com/pyscf/pyscf`.

@sunqm or @fishjojo, what's the process to add the CMakeLists.txt file to the sources tar file on pypi?

**Comment by hebrewsnabla** [comment]
Maybe a temporary lazy workaround can be: just specifying the 'requires_python' in pypi to prevent any install under py3.10, before the 3.10 wheel is added. That won't affect pip install via git, i think.

**Comment by chetmurthy** [comment]
I just created a fresh Conda environment with python=3.10.4, executed `pip install pyscf` and got this same error.  I then executed `pip cache purge` to purge the cache, retried the install, and (after downloading the tarball) it failed in the same way.

Interestingly, I also cloned the git repo, did `git checkout v2.0.1` and `pip install -e .` is happily progressing (lots of C compiles).  Perhaps the tarball needs fixing.

**Comment by AimAlee** [comment]
(avisofes) C:\Users\hp\build>cmake .. -DCMAKE_INSTALL_PREFIX=/C:\Users\hp\build
CMake Warning:
  Ignoring extra path from command line:

   ".."


CMake Error: The source directory "C:/Users/hp" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.



I am trying to run these commands... please help
Building & Installing

One nice and highly recommended feature of CMake is the ability to do out of source builds. In this way you can make all your .o files, various temporary depend files, and even the binary executables without cluttering up your source tree. To use out of source builds, first create a build directory, then change into your build directory and run cmake pointing it to the source directory:

mkdir build && cd build
cmake .. -DCMAKE_INSTALL_PREFIX=/opt/local

https://github.com/CNES/aviso-fes/blob/main/INSTALL.md

**Comment by AimAlee** [comment]
> 

Facing same problem using anaconda on windows 10. Please help

**Comment by AimAlee** [comment]
> > (avisofes) C:\Users\hp\build>cmake .. -DCMAKE_INSTALL_PREFIX=/C:\Users\hp\build CMake Warning: Ignoring extra path from command line:
> > ".."
> > CMake Error: The source directory "C:/Users/hp" does not appear to contain CMakeLists.txt. Specify --help for usage, or press the help button on the CMake GUI.
> > I am trying to run these commands... please help Building & Installing
> > One nice and highly recommended feature of CMake is the ability to do out of source builds. In this way you can make all your .o files, various temporary depend files, and even the binary executables without cluttering up your source tree. To use out of source builds, first create a build directory, then change into your build directory and run cmake pointing it to the source directory:
> > mkdir build && cd build cmake .. -DCMAKE_INSTALL_PREFIX=/opt/local
> > https://github.com/CNES/aviso-fes/blob/main/INSTALL.md
> 
> Windows is not supported. See [#1112 (comment)](https://github.com/pyscf/pyscf/issues/1112#issuecomment-972498362)

which windows would be compatible please?

**Comment by AimAlee** [comment]
Is there any association of following with my problem please?


https://www.scivision.dev/cmake-findmpi-prefix/
We discovered a long-standing [bug](https://gitlab.kitware.com/cmake/cmake/-/issues/23375) in CMake find_package(MPI) location hint via [CMAKE_PREFIX_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html) variable or environment variable [CMAKE_PREFIX_PATH](https://cmake.org/cmake/help/latest/envvar/CMAKE_PREFIX_PATH.html?highlight=cmake_prefix_path). For an unknown number of releases, CMake FindMPI ignores CMAKE_PREFIX_PATH, which normally takes a high priority in the search order. To workaround this issue, use variable MPI_ROOT to tell FindMPI where to look for MPI.

**Comment by xubwa** [comment]
@AimAlee No windows version is supported, but you can try windows subsystem of Linux (wsl) instead.

**Comment by sunqm** [comment]
Fixed in version 2.1
